### PR TITLE
Switch statement not correctly translated when default case is not present

### PR DESCRIPTION
**Pull request by github-sajan**
…esent

### CODE REVIEW

**Comment by mbudiu-vmw** [comment]
Why isn't the default case present? Previous passes should have inserted one.

**Review by mbudiu-vmw — APPROVED** [review]

**Comment by github-sajan** [comment]
> Why isn't the default case present? Previous passes should have inserted one.



> Why isn't the default case present? Previous passes should have inserted one.

Currently, in "SwitchAddDefault" pass in frontend, default entry in switch is only inserted if the expression is based on "Action Type" or "action_run". There is no implementation for adding default entry for generic switch statement with integer or enumerated type expression. 

So, as processing for switch statement with integer or enumerated type expression is done in "EliminateSwitch" pass in midend, implementation for adding default entry has been added here if it is not present

**Comment by mbudiu-vmw** [comment]
I have already approved this, but you still have conflicts that you need to rebase.

**Comment by github-sajan** [comment]
> I have already approved this, but you still have conflicts that you need to rebase.

I have resolved the conflict. Kindly approve and merge

File: testdata/p4_14_samples_outputs/copy_to_cpu-first.p4 [hunk 1]
```diff
@@ -26,9 +26,9 @@ struct metadata {
 }
 
 struct headers {
-    @name(".cpu_header")
+    @name(".cpu_header")
```
Comment (mbudiu-vmw) [inline testdata/p4_14_samples_outputs/copy_to_cpu-first.p4:29]: where do these extra spaces come from?
something is not right, they should not be here.

**Comment by mbudiu-vmw** [comment]
If the extra spaces appeared after you rebased there is a bug somewhere else in the compiler that we should track.

File: testdata/p4_14_samples_outputs/copy_to_cpu-first.p4 (hunk 1 above)
Comment (usha1830) [reply testdata/p4_14_samples_outputs/copy_to_cpu-first.p4:29]: Last change to these reference files seems to be coming from 
https://github.com/p4lang/p4c/pull/2902

Comment (mbudiu-vmw) [reply testdata/p4_14_samples_outputs/copy_to_cpu-first.p4:29]: But that change seems to remove the spaces, whereas you seem to add them

Comment (github-sajan) [reply testdata/p4_14_samples_outputs/copy_to_cpu-first.p4]: I have reverted back the code pushed which included space and pushed the code again. Please check and let me know if any conflicts are still there

**Comment by github-sajan** [comment]
> If the extra spaces appeared after you rebased there is a bug somewhere else in the compiler that we should track.

I have raised a new PR for the same and closing this out due to other irrelevant files getting updated while resolving conflict and rebasing. Kindly check and approve for the same
 https://github.com/p4lang/p4c/pull/3002

**Pull request closed without merging**